packages/core/platform/platform.js

```diff
@@ -7,38 +8,41 @@
  * @public
  */
 
-import uniq from 'ramda/src/uniq';
-
 import deprecate from '../internal/deprecate';
 
+// Refer the following for more details: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis
+const browserEnvironment = () => !!globalThis.window;
+
+// Note: Always true for Chrome v70 or higher: https://chromestatus.com/feature/4764225348042752
+const featureTouchEvent = () => browserEnvironment() && ('TouchEvent' in globalThis);
+
+// Refer https://patrickhlauke.github.io/touch/touchscreen-detection/
+const deviceTouchScreen = () => browserEnvironment() && (
+	// check if maxTouchPoints is greater than 0 first
+	globalThis.navigator?.maxTouchPoints > 0 ||
+	// check for any-pointer: coarse which mostly means touchscreen
+	globalThis.matchMedia?.("(any-pointer: coarse)")?.matches
+);
+
+/*
+ * Legacy Code to be removed in the next major release
+ */
 const hasGesture = () => {
 	return Boolean(
-		('ongesturestart' in window) ||
-		('onmsgesturestart' in window && (
-			window.navigator.msMaxTouchPoints > 1 ||
+		('ongesturestart' in window) || // non-standard and no major browser supports gesture events
+		('onmsgesturestart' in window && ( // Internet Explorer 10 only
+			window.navigator.msMaxTouchPoints > 1 || // Internet Explorer 10 only
 			window.navigator.maxTouchPoints > 1
 		))
 	);
 };
 
 const hasTouch = () => {
 	return Boolean(
-		('TouchEvent' in window) ||
-		('ontouchstart' in window) ||
-		window.navigator.msMaxTouchPoints ||
-		(window.navigator.msManipulationViewsEnabled && window.navigator.maxTouchPoints)
-	);
-};
-
-// Adapted from https://patrickhlauke.github.io/touch/touchscreen-detection/
-// I've omitted the touch event fallback since that is covered by hasTouch and we're less concerned
-// with legacy browsers used in touchscreen environments.
-const hasTouchScreen = () => {
-	return (
-		// if Pointer Events are supported, just check maxTouchPoints
-		(window.PointerEvent && ('maxTouchPoints' in window.navigator) && window.navigator.maxTouchPoints > 0) ||
-		// check for any-pointer:coarse which mostly means touchscreen
-		(window.matchMedia && window.matchMedia('(any-pointer:coarse)').matches)
+		featureTouchEvent() ||
```

Comment:
Do we need to update `hasTouch` logic which will be deprecated with featureTouchEvent()?

Reply:
`hasTouch` will be removed on the next major version update, but `featureTouchEvent` will remain.